### PR TITLE
Strategies: Stochastic MACD

### PR DESCRIPTION
**Pull request by talvasconcelos**
This is the first time i make a pull request, i'm not sure if this is the correct way to do it. If there's something that i'm doing wrong please correct me. There's 7 files changed because i was playing around with some files and GIT included them here and i don't know how to remove them from the PR.

This is a strategy i made, Stochastic MACD. Maybe it needs some code review from some of the bot experts. 

Basically this is a 2 step strategy, get trend direction with MACD, then determine entry point with Stochastic Oscillator.

**Buy signal**
1 - MACD is 0 or higher
2 - SRSI line crosses oversold line from the bottom-up

**Sell signal**
1 - MACD is lower than 0
2 - SRSI line crosses overbought line from the above-down

Run a couple simulations with it, with selector poloniex.ETC-BTC got 2% gain vs buy hold with the default values. DOGE-BTC also did positive despite the sideways nature of the market. also tried BTC-USDT with positive results

### CODE REVIEW

File: extensions/strategies/trend_ema/strategy.js [hunk 1]
```diff
@@ -53,15 +53,15 @@ module.exports = function container (get, set, clear) {
...
+          s.signal = !s.acted_on_trend ? 'sell' : null
           s.cancel_down = false
         }
         else if (!s.cancel_down && s.period.trend_ema_rate < (s.period.trend_ema_stddev * -1)) {
           if (s.trend !== 'down') {
             s.acted_on_trend = false
           }
           s.trend = 'down'
-          s.signal = !s.acted_on_trend ? 'sell' : null
+          s.signal = !s.acted_on_trend ? 'buy' : null
```
Comment (DeviaVir) [inline extensions/strategies/trend_ema/strategy.js]: why is this included in your PR?

**Comment by talvasconcelos** [comment]
it shouldn't be included, i altered the file locally but wasn't to be
included. first PR I've ever made.

**Review by DeviaVir — APPROVED** [review]

**Comment by DeviaVir** [comment]
Thanks for doing the legwork on this one! I made a few small adjustments and cleaned it up slightly (please review those changes) :+1:
